tests/manage/z_cluster/nodes/test_toleration.py

```diff
@@ -0,0 +1,61 @@
+import logging
+import pytest
+from ocs_ci.framework.testlib import (
+    tier4a,
+    E2ETest,
+)
+from ocs_ci.ocs import defaults
+from ocs_ci.ocs.resources.pod import (
+    get_all_pods,
+    check_toleration_on_pods,
+    delete_pods,
+    validate_pods_are_respinned_and_running_state,
+    wait_for_pods_to_be_running,
+)
+from ocs_ci.ocs.node import taint_nodes, untaint_nodes
+
+
+logger = logging.getLogger(__name__)
+
+
+@tier4a
+@pytest.mark.polarion_id("OCS-2450")
+class TestTaintAndTolerations(E2ETest):
+    """
+    Test to test taints and toleration
+    """
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        """
+        Make sure all nodes are untainted again
+
+        """
+
+        def finalizer():
+            untaint_nodes()
```

Comment:
You are not throwing any exceptions in the function, so though untaint fail we may not notice it in the test script.